src/file-events.c

```diff
@@ -196,17 +198,20 @@ static __always_inline void exit_mkdir(struct pt_regs *ctx, tail_call_slot_t tai
     fm->u.action.buffer_len = sizeof(file_message_t);
     fm->u.action.u.create.source_link = LINK_NONE;
 
-    cached_path->next_dentry = event.target_dentry;
+    init_filtered_cached_path(cached_path, event.target_dentry, event.target_vfsmount);
 
     ResolveName:
     ret = write_path(ctx, cached_path, &cursor, tail_call);
-    cached_path->next_dentry = NULL;
```

Comment:
I think we should keep this here. I'd consider it safer to reset `next_dentry` immediately. Otherwise in the code as it is written right now it the case of an error in `write_path` it doesn't look like we are resetting `next_dentry` so the next time this CPU loads `percpu_path` it will still have a `next_dentry` in there.

Reply:
There are only 2 `return` statements reachable after `write_path` returns and they both reset `next_dentry` to NULL before returning.  The `EventMismatch`, `EmitWarning`, and `NoEvent` labels all exit at the same point.